### DISCUSSION TITLE
E-commerce solution

### DISCUSSION THREAD

**Discussion by MrCasquette** [directus/directus#16778] [Archive]
Hey y'all Directus team,
You always have great answers to my questions. So I have another today. I'm pretty sure you'll find the best solution. I try to create an e commerce with directus because I love directus so much. But for the moment I only find a solution with Stripe and a personal Cart and another solution with Snipcart. Also find openCart and even't tried it. But all these pretty solutions add some dashboards and tasks to my clients. Is there a solution to make an entire e-commerce with directus with a secured way and implementing payment and shipping / delivering, without having a full skills of security development. I'm pretty noob with security. I hope I'm clear. To summarize, I just want to make an e-commerce with directus (pretty easy) but how to manage cart and everything we need as a store. 

I already know how to create products, categories, customers, even bill etc...

But how to link everything with payment and shipping securely is there a common way with directus?

Thanks for the answer I hope making all my projects with directus 😍

**Comment by paescuj** [comment]
A fully-fledged e-commerce solution which is completely based on Directus is certainly possible, but means a great effort. This actually applies to all custom e-commerce solutions as it requires a lot of custom logic (payment, shipping, tax calculation, ...)!

Most likely, the way to go would be to create a set of [extensions](https://docs.directus.io/extensions/introduction.html) which serve as endpoints for things like payments and shopping card as well as some hooks which do calculations and similar things.

As you described yourself, you could integrate some external components in your frontend (Stripe, Snipcart, ...) and use Directus mainly for the content (products, categories, customer, ...).

Another possibility is a hybrid setup, where Directus is again mainly used for content and another software serves as the e-commerce backend.

All in all, it heavily depends on your requirements. For example, if you don't need to do tax calculations or you handle payments manually, I think a solution based entirely on Directus can be implemented significantly faster.

You might also be interested in [existing discussions](https://github.com/directus/directus/discussions?discussions_q=e-commerce) around this topic.

**Reply by MrCasquette** [reply]
Thank you, I don't know why I wasn't informed of your answer. I now understand that I need to go further in learning about extensions, as I have never had to use them. Thanks again :)

**Reply by paescuj** [reply]
Hmm, indeed strange... Sounds like a good plan 👍 Much success!

**Reply by vitalijalbu** [reply]
Hi guys, I've read somewhere that for the cart system you can use medusa js... then no more need directus??

**Reply by pavelstianko** [reply]
I also found Medusa JS interesting for headless ecommerce. Also there are plugins for CMS: https://medusajs.com/plugins/?filters=CMS - it would be interesting to try combinate Medusa API and Directus :)

**Comment by nexthis** [comment]
I try, to make some stripe extensions https://github.com/nexthis/directus-stripe. If it were already working properly, I think the rest of the logic could be done using flow

**Reply by MrCasquette** [reply]
Thank for the share I'll take a look when I get some time ^^
